storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestDefaultResourceAwareStrategy.java

```diff
@@ -628,7 +628,7 @@ public void testMultipleRacksWithFavoritism() {
         t1Conf.putAll(config);
         final List<String> t1FavoredHostNames = Arrays.asList("host-41", "host-42", "host-43");
         t1Conf.put(Config.TOPOLOGY_SCHEDULER_FAVORED_NODES, t1FavoredHostNames);
-        final List<String> t1UnfavoredHostIds = Arrays.asList("host-1", "host-2", "host-3");
+        final List<String> t1UnfavoredHostIds = Arrays.asList("host-1", "host-2", "host-3", "host[11-13]", "host[2-1]");
```

Comment:
Could we add an assert to check the correct host ids were unfavored?

Reply:
I ran the local test but it is weird that none of the host ids in topo1 favored list show up in the assigned cluster. 
The original test code didn't test favored or unfavored hostIds at all.
```
        for (WorkerSlot ws : assignment.getSlotToExecutors().keySet()) {
            String hostName = rs.idToNode(ws.getNodeId()).getHostname();
            String rackId = resolvedSuperVisors.get(hostName);
            Assert.assertTrue(ws + " is neither on a favored node " + t1FavoredHostNames + " nor the highest priority rack (rack-0)",
                t1FavoredHostNames.contains(hostName) || "rack-0".equals(rackId));
            Assert.assertFalse(ws + " is a part of an unfavored node " + t1UnfavoredHostIds,
                t1UnfavoredHostIds.contains(hostName));
        }
```
The assigned hostIds are always host-001, host-000, host-002, host-003. Is that expected in local unit test?

@revans2 , since you wrote the test code, do you have any insights about the issue?